### DISCUSSION TITLE
How to handle ZodIssue type?

### DISCUSSION THREAD

**Discussion by IRediTOTO** [colinhacks/zod#1318] [Q&A]
Hi, I am trying to show the error message from Zod, but how can I hadnle the type here? I know the error object return have field `received` but how to check it ?
![image](https://user-images.githubusercontent.com/15695252/183267459-64c27612-f11d-4a82-89d3-8e7a33081f1b.png)

**Comment by nullndr** [comment]
`received` does not exists in `ZodIssue`: see [here](https://github.com/colinhacks/zod/blob/d506b083d4a7b2e5600317bd44ca1dba97709fe8/ERROR_HANDLING.md#zodissue).

If you use `safeParse` you can easily get it like this:

```ts
const data = z.number().safeParse(1)

if(data.success) {
  // all good
} else {
  console.log(data.error)
}
```
